### ISSUE TITLE
Implementing an interface with <any> should not raise an unsafe type warning

### ISSUE THREAD

**Issue by LHSnow** [SonarSource/SonarTS#601]
**I want to report a bug.**
When implementing interfaces, it should be allowed to use the otherwise unsafe type `any`. As the interface is in charge of the expected types, the implementing class should not be punished for following them. 

**Node.js version:** 8.10.0 <br>
**TypeScript version:**  2.6.2 <br>
**SonarQube version:** 6.7 (build 33306) <br>
**TSLint version:** 5.7.0 <br>
**Rule key:** S4204 

**Reproducer** 
```
//ControlValueAccessor is an Angular interface for creating new form components
export class CustomFormComponent implements ControlValueAccessor {
  writeValue(value: any): void {  }

  registerOnChange(fn: (value: any) => void): void {  }

  registerOnTouched(fn: () => void): void {  }
}
```
Reference: https://angular.io/api/forms/ControlValueAccessor

**Comment by vilchik-elena** [comment]
Why can't you use more specific type in the class implementing `ControlValueAccessor`:

```javascript
export class CustomFormComponent implements ControlValueAccessor {
  writeValue(value: number): void {  }
 // ...
}
```

**Comment by LHSnow** [comment]
You are right, coming from a Java background where your example would not count as overriding the method, I wrongly assumed this would not work in TS either. I am not sure if there are other examples around, that would keep this issue open, but for my part it can be closed as a non-fix. I guess we can always re-open this if valid examples turn up.
